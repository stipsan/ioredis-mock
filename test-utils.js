// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [commandName, `${commandName}Buffer`]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(command, (a, b) => a.equals(Buffer.from(b)))
  )
}
