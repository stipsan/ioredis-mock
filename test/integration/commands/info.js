import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { runTwinSuite } from '../../../test-utils'

runTwinSuite('info', command => {
  describe(command, () => {
    it('should return default info with CRLF', async () => {
      const redis = new Redis()
      let value = await redis[command]()
      if (command === 'infoBuffer') {
        expect(Buffer.isBuffer(value)).toBe(true)
        value = value.toString()
      }
      const lines = value.split('\r\n')

      expect(lines).toHaveLength(value.split('\n').length)
      expect(lines.some(line => line.startsWith('redis_version:'))).toBeTruthy()
      redis.disconnect()
    })
  })
})
