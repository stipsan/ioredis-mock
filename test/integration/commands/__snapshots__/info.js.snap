// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info should return default info with CRLF 1`] = `"redis_version:7.0.10"`;

exports[`info should return enough info to be useful 1`] = `
[
  "# Server
",
  "redis_version:7.0.10
",
  "redis_git_sha1:00000000
",
  "redis_git_dirty:0
",
  "redis_build_id:f03ae0273e39f2e3
",
  "redis_mode:standalone
",
  "os:Linux 5.15.0-1034-azure x86_64
",
  "arch_bits:64
",
  "monotonic_clock:POSIX clock_gettime
",
  "multiplexing_api:epoll
",
  "atomicvar_api:c11-builtin
",
  "gcc_version:10.2.1
",
  "process_id:1
",
  "process_supervised:no
",
  "run_id:f3376b877332b738fbf3fa35a357aad5a560bed5
",
  "tcp_port:6379
",
  "server_time_usec:1679518476581348
",
  "uptime_in_seconds:63
",
  "uptime_in_days:0
",
  "hz:10
",
  "configured_hz:10
",
  "lru_clock:1796876
",
  "executable:/tmp/redis-server
",
  "config_file:
",
  "io_threads_active:0
",
  "
",
  "# Clients
",
  "connected_clients:1
",
  "cluster_connections:0
",
  "maxclients:10000
",
  "client_recent_max_input_buffer:20480
",
  "client_recent_max_output_buffer:0
",
  "blocked_clients:0
",
  "tracking_clients:0
",
  "clients_in_timeout_table:0
",
  "
",
  "# Memory
",
  "used_memory:3399672
",
  "used_memory_human:3.24M
",
  "used_memory_rss:14741504
",
  "used_memory_rss_human:14.06M
",
  "used_memory_peak:3530144
",
  "used_memory_peak_human:3.37M
",
  "used_memory_peak_perc:96.30%
",
  "used_memory_overhead:863168
",
  "used_memory_startup:862048
",
  "used_memory_dataset:2536504
",
  "used_memory_dataset_perc:99.96%
",
  "allocator_allocated:3590472
",
  "allocator_active:4362240
",
  "allocator_resident:6955008
",
  "total_system_memory:7281278976
",
  "total_system_memory_human:6.78G
",
  "used_memory_lua:39936
",
  "used_memory_vm_eval:39936
",
  "used_memory_lua_human:39.00K
",
  "used_memory_scripts_eval:936
",
  "number_of_cached_scripts:4
",
  "number_of_functions:0
",
  "number_of_libraries:0
",
  "used_memory_vm_functions:32768
",
  "used_memory_vm_total:72704
",
  "used_memory_vm_total_human:71.00K
",
  "used_memory_functions:184
",
  "used_memory_scripts:1120
",
  "used_memory_scripts_human:1.09K
",
  "maxmemory:0
",
  "maxmemory_human:0B
",
  "maxmemory_policy:noeviction
",
  "allocator_frag_ratio:1.21
",
  "allocator_frag_bytes:771768
",
  "allocator_rss_ratio:1.59
",
  "allocator_rss_bytes:2592768
",
  "rss_overhead_ratio:2.12
",
  "rss_overhead_bytes:7786496
",
  "mem_fragmentation_ratio:4.39
",
  "mem_fragmentation_bytes:11381704
",
  "mem_not_counted_for_evict:0
",
  "mem_replication_backlog:0
",
  "mem_total_replication_buffers:0
",
  "mem_clients_slaves:0
",
  "mem_clients_normal:0
",
  "mem_cluster_links:0
",
  "mem_aof_buffer:0
",
  "mem_allocator:jemalloc-5.2.1
",
  "active_defrag_running:0
",
  "lazyfree_pending_objects:0
",
  "lazyfreed_objects:0
",
  "
",
  "# Persistence
",
  "loading:0
",
  "async_loading:0
",
  "current_cow_peak:0
",
  "current_cow_size:0
",
  "current_cow_size_age:0
",
  "current_fork_perc:0.00
",
  "current_save_keys_processed:0
",
  "current_save_keys_total:0
",
  "rdb_changes_since_last_save:0
",
  "rdb_bgsave_in_progress:0
",
  "rdb_last_save_time:1679518476
",
  "rdb_last_bgsave_status:ok
",
  "rdb_last_bgsave_time_sec:0
",
  "rdb_current_bgsave_time_sec:-1
",
  "rdb_saves:3
",
  "rdb_last_cow_size:286720
",
  "rdb_last_load_keys_expired:0
",
  "rdb_last_load_keys_loaded:0
",
  "aof_enabled:0
",
  "aof_rewrite_in_progress:0
",
  "aof_rewrite_scheduled:0
",
  "aof_last_rewrite_time_sec:0
",
  "aof_current_rewrite_time_sec:-1
",
  "aof_last_bgrewrite_status:ok
",
  "aof_rewrites:1
",
  "aof_rewrites_consecutive_failures:0
",
  "aof_last_write_status:ok
",
  "aof_last_cow_size:212992
",
  "module_fork_in_progress:0
",
  "module_fork_last_cow_size:0
",
  "
",
  "# Stats
",
  "total_connections_received:800
",
  "total_commands_processed:2367
",
  "instantaneous_ops_per_sec:63
",
  "total_net_input_bytes:60777
",
  "total_net_output_bytes:4033522
",
  "total_net_repl_input_bytes:0
",
  "total_net_repl_output_bytes:0
",
  "instantaneous_input_kbps:1.21
",
  "instantaneous_output_kbps:134.29
",
  "instantaneous_input_repl_kbps:0.00
",
  "instantaneous_output_repl_kbps:0.00
",
  "rejected_connections:0
",
  "sync_full:0
",
  "sync_partial_ok:0
",
  "sync_partial_err:0
",
  "expired_keys:16
",
  "expired_stale_perc:1.82
",
  "expired_time_cap_reached_count:0
",
  "expire_cycle_cpu_milliseconds:0
",
  "evicted_keys:0
",
  "evicted_clients:0
",
  "total_eviction_exceeded_time:0
",
  "current_eviction_exceeded_time:0
",
  "keyspace_hits:325
",
  "keyspace_misses:102
",
  "pubsub_channels:0
",
  "pubsub_patterns:0
",
  "pubsubshard_channels:0
",
  "latest_fork_usec:230
",
  "total_forks:2
",
  "migrate_cached_sockets:0
",
  "slave_expires_tracked_keys:0
",
  "active_defrag_hits:0
",
  "active_defrag_misses:0
",
  "active_defrag_key_hits:0
",
  "active_defrag_key_misses:0
",
  "total_active_defrag_time:0
",
  "current_active_defrag_time:0
",
  "tracking_total_keys:0
",
  "tracking_total_items:0
",
  "tracking_total_prefixes:0
",
  "unexpected_error_replies:0
",
  "total_error_replies:66
",
  "dump_payload_sanitizations:0
",
  "total_reads_processed:3168
",
  "total_writes_processed:2373
",
  "io_threaded_reads_processed:0
",
  "io_threaded_writes_processed:0
",
  "reply_buffer_shrinks:24
",
  "reply_buffer_expands:0
",
  "
",
  "# Replication
",
  "role:master
",
  "connected_slaves:0
",
  "master_failover_state:no-failover
",
  "master_replid:6640fcb3e02fb20344185a04862b640380705f03
",
  "master_replid2:0000000000000000000000000000000000000000
",
  "master_repl_offset:0
",
  "second_repl_offset:-1
",
  "repl_backlog_active:0
",
  "repl_backlog_size:1048576
",
  "repl_backlog_first_byte_offset:0
",
  "repl_backlog_histlen:0
",
  "
",
  "# CPU
",
  "used_cpu_sys:0.250358
",
  "used_cpu_user:0.196469
",
  "used_cpu_sys_children:0.000918
",
  "used_cpu_user_children:0.004110
",
  "used_cpu_sys_main_thread:0.249107
",
  "used_cpu_user_main_thread:0.195822
",
  "
",
  "# Modules
",
  "
",
  "# Errorstats
",
  "errorstat_ERR:count=50
",
  "errorstat_NOSCRIPT:count=2
",
  "errorstat_WRONGTYPE:count=14
",
  "
",
  "# Cluster
",
  "cluster_enabled:0
",
  "
",
  "# Keyspace
",
  "",
]
`;

exports[`info should return the Redis version number used during end-to-end testing 1`] = `
"redis_version:7.0.10
"
`;

exports[`infoBuffer should return default info with CRLF 1`] = `"redis_version:7.0.10"`;

exports[`infoBuffer should return enough info to be useful 1`] = `
[
  "# Server
",
  "redis_version:7.0.10
",
  "redis_git_sha1:00000000
",
  "redis_git_dirty:0
",
  "redis_build_id:f03ae0273e39f2e3
",
  "redis_mode:standalone
",
  "os:Linux 5.15.0-1034-azure x86_64
",
  "arch_bits:64
",
  "monotonic_clock:POSIX clock_gettime
",
  "multiplexing_api:epoll
",
  "atomicvar_api:c11-builtin
",
  "gcc_version:10.2.1
",
  "process_id:1
",
  "process_supervised:no
",
  "run_id:f3376b877332b738fbf3fa35a357aad5a560bed5
",
  "tcp_port:6379
",
  "server_time_usec:1679518476581348
",
  "uptime_in_seconds:63
",
  "uptime_in_days:0
",
  "hz:10
",
  "configured_hz:10
",
  "lru_clock:1796876
",
  "executable:/tmp/redis-server
",
  "config_file:
",
  "io_threads_active:0
",
  "
",
  "# Clients
",
  "connected_clients:1
",
  "cluster_connections:0
",
  "maxclients:10000
",
  "client_recent_max_input_buffer:20480
",
  "client_recent_max_output_buffer:0
",
  "blocked_clients:0
",
  "tracking_clients:0
",
  "clients_in_timeout_table:0
",
  "
",
  "# Memory
",
  "used_memory:3399672
",
  "used_memory_human:3.24M
",
  "used_memory_rss:14741504
",
  "used_memory_rss_human:14.06M
",
  "used_memory_peak:3530144
",
  "used_memory_peak_human:3.37M
",
  "used_memory_peak_perc:96.30%
",
  "used_memory_overhead:863168
",
  "used_memory_startup:862048
",
  "used_memory_dataset:2536504
",
  "used_memory_dataset_perc:99.96%
",
  "allocator_allocated:3590472
",
  "allocator_active:4362240
",
  "allocator_resident:6955008
",
  "total_system_memory:7281278976
",
  "total_system_memory_human:6.78G
",
  "used_memory_lua:39936
",
  "used_memory_vm_eval:39936
",
  "used_memory_lua_human:39.00K
",
  "used_memory_scripts_eval:936
",
  "number_of_cached_scripts:4
",
  "number_of_functions:0
",
  "number_of_libraries:0
",
  "used_memory_vm_functions:32768
",
  "used_memory_vm_total:72704
",
  "used_memory_vm_total_human:71.00K
",
  "used_memory_functions:184
",
  "used_memory_scripts:1120
",
  "used_memory_scripts_human:1.09K
",
  "maxmemory:0
",
  "maxmemory_human:0B
",
  "maxmemory_policy:noeviction
",
  "allocator_frag_ratio:1.21
",
  "allocator_frag_bytes:771768
",
  "allocator_rss_ratio:1.59
",
  "allocator_rss_bytes:2592768
",
  "rss_overhead_ratio:2.12
",
  "rss_overhead_bytes:7786496
",
  "mem_fragmentation_ratio:4.39
",
  "mem_fragmentation_bytes:11381704
",
  "mem_not_counted_for_evict:0
",
  "mem_replication_backlog:0
",
  "mem_total_replication_buffers:0
",
  "mem_clients_slaves:0
",
  "mem_clients_normal:0
",
  "mem_cluster_links:0
",
  "mem_aof_buffer:0
",
  "mem_allocator:jemalloc-5.2.1
",
  "active_defrag_running:0
",
  "lazyfree_pending_objects:0
",
  "lazyfreed_objects:0
",
  "
",
  "# Persistence
",
  "loading:0
",
  "async_loading:0
",
  "current_cow_peak:0
",
  "current_cow_size:0
",
  "current_cow_size_age:0
",
  "current_fork_perc:0.00
",
  "current_save_keys_processed:0
",
  "current_save_keys_total:0
",
  "rdb_changes_since_last_save:0
",
  "rdb_bgsave_in_progress:0
",
  "rdb_last_save_time:1679518476
",
  "rdb_last_bgsave_status:ok
",
  "rdb_last_bgsave_time_sec:0
",
  "rdb_current_bgsave_time_sec:-1
",
  "rdb_saves:3
",
  "rdb_last_cow_size:286720
",
  "rdb_last_load_keys_expired:0
",
  "rdb_last_load_keys_loaded:0
",
  "aof_enabled:0
",
  "aof_rewrite_in_progress:0
",
  "aof_rewrite_scheduled:0
",
  "aof_last_rewrite_time_sec:0
",
  "aof_current_rewrite_time_sec:-1
",
  "aof_last_bgrewrite_status:ok
",
  "aof_rewrites:1
",
  "aof_rewrites_consecutive_failures:0
",
  "aof_last_write_status:ok
",
  "aof_last_cow_size:212992
",
  "module_fork_in_progress:0
",
  "module_fork_last_cow_size:0
",
  "
",
  "# Stats
",
  "total_connections_received:800
",
  "total_commands_processed:2367
",
  "instantaneous_ops_per_sec:63
",
  "total_net_input_bytes:60777
",
  "total_net_output_bytes:4033522
",
  "total_net_repl_input_bytes:0
",
  "total_net_repl_output_bytes:0
",
  "instantaneous_input_kbps:1.21
",
  "instantaneous_output_kbps:134.29
",
  "instantaneous_input_repl_kbps:0.00
",
  "instantaneous_output_repl_kbps:0.00
",
  "rejected_connections:0
",
  "sync_full:0
",
  "sync_partial_ok:0
",
  "sync_partial_err:0
",
  "expired_keys:16
",
  "expired_stale_perc:1.82
",
  "expired_time_cap_reached_count:0
",
  "expire_cycle_cpu_milliseconds:0
",
  "evicted_keys:0
",
  "evicted_clients:0
",
  "total_eviction_exceeded_time:0
",
  "current_eviction_exceeded_time:0
",
  "keyspace_hits:325
",
  "keyspace_misses:102
",
  "pubsub_channels:0
",
  "pubsub_patterns:0
",
  "pubsubshard_channels:0
",
  "latest_fork_usec:230
",
  "total_forks:2
",
  "migrate_cached_sockets:0
",
  "slave_expires_tracked_keys:0
",
  "active_defrag_hits:0
",
  "active_defrag_misses:0
",
  "active_defrag_key_hits:0
",
  "active_defrag_key_misses:0
",
  "total_active_defrag_time:0
",
  "current_active_defrag_time:0
",
  "tracking_total_keys:0
",
  "tracking_total_items:0
",
  "tracking_total_prefixes:0
",
  "unexpected_error_replies:0
",
  "total_error_replies:66
",
  "dump_payload_sanitizations:0
",
  "total_reads_processed:3168
",
  "total_writes_processed:2373
",
  "io_threaded_reads_processed:0
",
  "io_threaded_writes_processed:0
",
  "reply_buffer_shrinks:24
",
  "reply_buffer_expands:0
",
  "
",
  "# Replication
",
  "role:master
",
  "connected_slaves:0
",
  "master_failover_state:no-failover
",
  "master_replid:6640fcb3e02fb20344185a04862b640380705f03
",
  "master_replid2:0000000000000000000000000000000000000000
",
  "master_repl_offset:0
",
  "second_repl_offset:-1
",
  "repl_backlog_active:0
",
  "repl_backlog_size:1048576
",
  "repl_backlog_first_byte_offset:0
",
  "repl_backlog_histlen:0
",
  "
",
  "# CPU
",
  "used_cpu_sys:0.250358
",
  "used_cpu_user:0.196469
",
  "used_cpu_sys_children:0.000918
",
  "used_cpu_user_children:0.004110
",
  "used_cpu_sys_main_thread:0.249107
",
  "used_cpu_user_main_thread:0.195822
",
  "
",
  "# Modules
",
  "
",
  "# Errorstats
",
  "errorstat_ERR:count=50
",
  "errorstat_NOSCRIPT:count=2
",
  "errorstat_WRONGTYPE:count=14
",
  "
",
  "# Cluster
",
  "cluster_enabled:0
",
  "
",
  "# Keyspace
",
  "",
]
`;

exports[`infoBuffer should return the Redis version number used during end-to-end testing 1`] = `
"redis_version:7.0.10
"
`;
